### ISSUE TITLE
Problem: The existing user-level documentation is way too limited

### ISSUE THREAD

**Issue by sirthias** [sirthias/swave#2]
Problem:
Users have very little information on

* the core entry points to the API
* the exact mechanics of each of the available stream operations
* the inner workings / limitations / guidelines

Additionally contributors don't have any (technical) information on the best process for contributing, especially with regard to adding new operations.

Proposed solution:
Put up http://swave.io and add proper documentation. :)

**Comment by fommil** [comment]
I saw down to read this tonight and was enjoying it until I hit the wave of TODO sections 😃

**Comment by sirthias** [comment]
@fommil Yes, clearly they need to go. However, now we have the proper structure in place to simply bang out those doc chapters. We expect to remove the last TODO before the end of November.

**Comment by fommil** [comment]
nice timing, was just telling some work colleagues how interesting swave is
